### PR TITLE
[Core] Modify SetValueOnIntegrationPoints to Python

### PR DESCRIPTION
**Pull request by tteschemacher**
Edit:
It allows conversion of vectors and python lists to arrays. This facilitate things from python as variables do not need to be defined as either vector or array which can be a bit complex.

### CODE REVIEW

**Review by loumalouomega — DISMISSED** [review]
👌

**Review by RiccardoRossi — DISMISSED** [review]
also blocking.

why a version without the s and one with the s?

**Comment by tteschemacher** [comment]
The one without it as it is the same in the code. The one with it, because I want to be kind to the persons that think you need it (like the editor of this).

It's annoying to have a bug due to that s...

File: kratos/python/add_mesh_to_python.cpp [hunk 1]
```diff
@@ -602,6 +611,11 @@ void  AddMeshToPython(pybind11::module& m)
     .def("SetValuesOnIntegrationPoints", SetValuesOnIntegrationPointsConstitutiveLaw)
     .def("SetValuesOnIntegrationPoints", SetValuesOnIntegrationPointsDouble<Element>)
     .def("SetValuesOnIntegrationPoints", SetValuesOnIntegrationPointsArray1d<Element>)
+    // SetValueOnIntegrationPoints
+    .def("SetValueOnIntegrationPoints", SetValuesOnIntegrationPointsVector<Element>)
+    .def("SetValueOnIntegrationPoints", SetValuesOnIntegrationPointsConstitutiveLaw)
+    .def("SetValueOnIntegrationPoints", SetValuesOnIntegrationPointsDouble<Element>)
+    .def("SetValueOnIntegrationPoints", SetValuesOnIntegrationPointsArray1d<Element>)
```
Comment (philbucher) [inline kratos/python/add_mesh_to_python.cpp]:
```suggestion
    .def("SetValuesOnIntegrationPoints", SetValuesOnIntegrationPointsArray1d<Element>)
```

this is proposed in #2194

Comment (tteschemacher) [reply kratos/python/add_mesh_to_python.cpp]: This is exactly what I don't want as it different as in the c++ code...

Comment (RiccardoRossi) [reply kratos/python/add_mesh_to_python.cpp]: my point is to have one ... so if in the C++ code is without s, i would leave everything without s

Comment (philbucher) [reply kratos/python/add_mesh_to_python.cpp]: ah ok, I didn't get this
So currently in C++ it is `SetValueOnIntegrationPoints` and in python it is `SetValuesOnIntegrationPoints`?

**Comment by tteschemacher** [comment]
I removed the name changes. I created #6517 to discuss the name issue. Thus, this is purely enhancing a feature for better usability.

**Comment by tteschemacher** [comment]
@RiccardoRossi do you agree on the modifications? Instead of adding additional python functions we purely add the possibility to create arrays from vectors and python lists. Makes use easier.

**Comment by tteschemacher** [comment]
@RiccardoRossi you're blocking this PR. Do you have any open concerns?

**Review by philbucher — APPROVED** [review]
ping @RiccardoRossi

**Review by RiccardoRossi — APPROVED** [review]
+1

**Pull request merged**